fili-core/src/main/java/com/yahoo/bard/webservice/exception/MetadataExceptionHandler.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.exception;
+
+import com.yahoo.bard.webservice.web.apirequest.ApiRequest;
+
+import java.util.Optional;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * Like {@link DataExceptionHandler}, but handles the metadata endpoints.
+ *
+ * The data and metadata endpoints have very different response models (asynchronous vs. synchronous respectively). As
+ * a result, there is not a clean way to unify them into a single exception handler. So they each
+ * get their own
+ */
+public interface MetadataExceptionHandler {
```

Comment:
Maybe we can have a marker interface `ExceptionHandler` whose child interfaces are `MetadataExceptionHandler ` and `DataExceptionHandler`

Reply:
That sounds like unnecessary infrastructure to me. Since both interfaces 
have such different signatures, we'll still need to either use the 
`(Meta)DataExceptionHandler` in the appropriate Servlet, or we'll have to cast.
So I don't know what such a marker interface buys us.